feature/migration/qrcode/qr-code-format.md

```diff
@@ -0,0 +1,463 @@
+
+### Writer
+
+A writer is the software creating QR codes that follow this specification, e.g. Thunderbird for desktop.
+
+### Account
+
+For the purpose of this specification an account is the combination of an `IncomingServer` object and the
+`OutgoingServerGroups` object that is directly following this `IncomingServer` object.
+See [root element](#root-element).
+
+### Version
+
+Whenever this document refers to a version without qualifier, the [specification version](#specification-versions) is
+meant. It is different from the [format version](#formatversion) that is expected to stay the same until a
+backward-incompatible change to the data format needs to be made.
+
+## General structure
+
+The QR code payload is JSON, encoded using UTF-8. The data objects like incoming server, outgoing server, and identities
+are mapped to JSON arrays.
+
+The data format is extensible by allowing arrays to contain additional elements from the ones specified in the initial
+version. This applies to all arrays unless otherwise stated, but it is usually also explicitly spelled out.
+For forward-compatibility a reader must ignore additional array values not listed in the version of the specification it
+implements. A writer must not add elements to an array that are not part of the version of the specification it
+implements.
+
+### Motivation
+
+We chose JSON because it's a widely supported format that is easily extensible, i.e. it's easy to add additional
+properties later. We decided to use nested JSON arrays because it leads to compact output and space in QR codes is very
+limited.
+
+The downside of using JSON arrays is that it's most likely more work for implementations. For most object oriented
+languages there exist libraries to map objects to JSON objects. However, library-assisted mapping of objects to JSON
+arrays is not something that is typically available.
+Another issue worth mentioning is that using JSON arrays makes the output harder to read for humans.
+
+This is not a great data format. But it's one that seems to work well within the constraints of a QR code.
+
+### Compatibility
+
+If you intend to create a reader that is not using all of the information present in the payload, you must still
+validate all of the properties, even the ones you're not using. This is to avoid situations where QR codes are being
+recognized as valid by one (incomplete) reader but not another (full) reader.
+
+## Data format
+
+### Root element
+
+- Available since: version 1
+- Type: Array
+
+The root element of the JSON document is an array containing the following elements in this order:
+
+- `FormatVersion`
+- `MiscellaneousData`
+- `IncomingServer`
+- `OutgoingServerGroups`
+
+The array may contain additional `IncomingServer` and `OutgoingServerGroup` elements. But they always have to appear in
+combination because only together they make up an account.
```

Comment:
```suggestion
pairs because only together they make up an account.
```
I think pairs makes it clearer that there always needs to be an `IncomingServer` and then an `OutgoingServerGroup` - actually, the order may still not be well specified.